### PR TITLE
Kim CNN: Early Stop Training (#40)

### PR DESCRIPTION
**Pull request by rosequ**
- Stop training if there's no improvement in the accuracy over 5 epochs
@Impavidity @HTAustin can you have a look at it?

### CODE REVIEW

File: kim_cnn/network/cnnTextNetwork.py [hunk 1]
```diff
@@ -152,6 +155,17 @@ def train(self):
 
       epoch += 1
       accuracy = float(acc_corrects) / float(acc_sents) * 100
+
+      # if the new accuracy is better than the old accuracy by 1e-3
+      if accuracy - best_accuracy > 0.01:
```
Comment (snapbug) [inline kim_cnn/network/cnnTextNetwork.py]: 1e-3 != 0.01, I suspect you meant 0.001?

Comment (rosequ) [reply kim_cnn/network/cnnTextNetwork.py]: Sorry, it should be 0.1 since I meant it in terms of percentage. I'll update the PR and the comment.

**Comment by snapbug** [comment]
Move the printing of the epoch accuracy to before the check and you'll get a 👍

**Comment by snapbug** [comment]
@lintool can this get merged in? It will make seatless experimentation a _lot_ easier.

**Pull request merged**